### PR TITLE
feat: improvements to `ShareModal` and its parsing

### PR DESCRIPTION
**Pull request by chicoferreira**
Rewrote ShareModal for a cleaner code and support for more formats.

Changes include:
- Now the name of the filter can be used instead of only the id
- Removed the loading animation when importing
- Changed nomenclature of "link" to "code"

Shifts parsing also keep the feature of parsing shifts separated by comma format (DSS=PL1,T1)

### CODE REVIEW

File: components/ShareModal/ShareModal.tsx [hunk 1]
```diff
@@ -227,61 +200,39 @@ const ShareModal = ({
...
-                      onSubmit={importEvents}
+                      onSubmit={importEventsHandle}
                     >
                       <div className="relative flex flex-grow items-stretch focus-within:z-10">
                         <input
-                          name="link"
+                          name="code"
                           className="block w-full rounded-none rounded-l-lg border-0 py-1.5 ring-1 ring-inset ring-neutral-300 placeholder:text-neutral-400 focus:ring-2 focus:ring-inset focus:ring-cesium-900 dark:bg-neutral-800 dark:ring-neutral-400/30 sm:text-sm sm:leading-6"
-                          placeholder="Insert a share link here..."
+                          placeholder="Insert a share link here"
```
Comment (diogogmatos) [inline components/ShareModal/ShareModal.tsx:209]: "link" -> "code"

**Review by diogogmatos — CHANGES_REQUESTED** [review]

**Review by diogogmatos — COMMENTED** [review]
`ADI=T1&ADI=PL1`
This code is being generated when selecting the T1 and PL1 shifts from the same subject, shouldn't they be formatted as `ADI=T1,PL1`? Also, that would be a nice refactor to the Calendarium Shifts API 👀

File: components/ShareModal/ShareModal.tsx [hunk 2]
```diff
@@ -291,14 +242,14 @@ const ShareModal = ({
                 <Collapse.Panel header="How does it work?" key="1">
                   <div className="text-justify">
                     <a className="font-medium">
-                      Copy your share code with <i className="bi bi-copy" />
+                      Copy your share code with <i className="bi bi-copy"/>
                     </a>{" "}
```
Comment (diogogmatos) [inline components/ShareModal/ShareModal.tsx:246]: `<a>` -> `<span>`
I know this was already there, but it was a mistake of mine 😅 This makes the browser think the text is a link and highlight it when hovering.

File: components/ShareModal/ShareModal.tsx [hunk 3]
```diff
@@ -291,14 +242,14 @@ const ShareModal = ({
...
                     to share your {isHome ? "events" : "schedule"} with your
                     friends or with another device.
                     <p></p>
                     <a className="font-medium">
                       Paste a share code and click{" "}
-                      <i className="bi bi-download" />
+                      <i className="bi bi-download"/>
                     </a>{" "}
```
Comment (diogogmatos) [inline components/ShareModal/ShareModal.tsx:253]: `<a>` -> `<span>`
Same here 🙏

**Review by diogogmatos — CHANGES_REQUESTED** [review]

**Comment by chicoferreira** [comment]
Shift copy output now has selected shifts with the same identifier grouped by commas

File: components/ShareModal/ShareModal.tsx [hunk 4]
```diff
@@ -18,183 +18,174 @@ type ShareModalProps = {
...
-    const link: string | File = formData.get("link");
-
-    if (link instanceof File) return false;
-
-    // for events
-    if (isHome) {
-      // get subject ids
-      const subjectIds: string[] = (link as string).split("&");
+    const rawCode = formData.get("code");
+    if (typeof "foo" !== "string") return false;
```
Comment (ruilopesm) [inline components/ShareModal/ShareModal.tsx:161]:
```suggestion
```

**Review by diogogmatos — APPROVED** [review]
LGTM! Consider applying the same technique you used in the parsing to allow for `ADI=T1&ADI=PL1` and `ADI=T1,PL1` in the Shifts Export API (`/api/export/shifts`), creating a new PR for that. Thank you! 🚀

**Pull request merged**